engine/src/main/java/org/terasology/world/propagation/AbstractFullWorldView.java

```diff
@@ -61,13 +63,14 @@ public byte getValueAt(Vector3i pos) {
      * @param pos   The internal position of the chunk to get the value from
      * @return The relevant value for this view
      */
-    protected abstract byte getValueAt(LitChunk chunk, Vector3i pos);
+    protected abstract byte getValueAt(LitChunk chunk, Vector3ic pos);
 
     @Override
-    public void setValueAt(Vector3i pos, byte value) {
-        setValueAt(getChunk(pos), ChunkMath.calcRelativeBlockPos(pos.x, pos.y, pos.z), value);
-        for (Vector3i affectedChunkPos : ChunkMath.getChunkRegionAroundWorldPos(pos, 1)) {
-            Chunk dirtiedChunk = chunkProvider.getChunk(affectedChunkPos);
+    public void setValueAt(Vector3ic pos, byte value) {
+        setValueAt(getChunk(pos), Chunks.toRelative(pos, new Vector3i()), value);
+        BlockRegion chunkRegion = new BlockRegion(pos).expand(1, 1, 1);
+        for (Vector3ic affectedChunkPos : Chunks.toChunkRegion(chunkRegion, chunkRegion)) {
```

Comment:
@pollend Can we offer `Chunks.toChunkRegion(blockRegion)` mutating the given region in-place as a short-hand for this? (as a follow-up, not blocking)

Reply:
I started a draft for this in #4389.